### PR TITLE
Add basic documentation

### LINKED ISSUE #280: This repo has no docs
There doesn't seem to be any documentation for `komodo`. The closest thing I could find is https://fmu-docs.equinor.com/docs/komodo/index.html (not a public link), which I think is really documentation for `komodo-releases`.

I think the documentation could contain the following:

- Expand on the basic example in the README, see #257 
- Explain how it relates to `komodoenv` and perhaps, with appropriate signals that it's not public, `komodo-releases` (likewise, those libraries should explain the relationship)
- Map the resulting release directory structure, files, etc.
- Explain all of the arguments to the `kmd` command, because most of them do not have expensive help in the CLI.
- Document the various commands: komodo-check-pypi, komodo-insert-proposals, komodo-post-messages, komodo-check-symlinks, komodo-lint, komodo-reverse-deps, komodo-clean-repository, komodo-lint-maturity, komodo-snyk-test, komodo-create-symlinks, komodo-lint-package-status,  komodo-suggest-symlinks, komodo-extract-dep-graph, komodo-non-deployed,         komodo-transpiler
- Show how (and if!) `komodo` is intended to be used as a library.

### PR DESCRIPTION
**Pull request by kwinkunks**
This package has no documentation; even docstrings are mostly missing. This commit adds basic Sphinx documentation using the Furo theme. It also adds a workflow to automatically build the docs on a release, or on a manual trigger. This action uses GitHub Pages and needs review.

PR resolves #280

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ versions. Each version contains:
 
 For example, we may have a `repository.yml` like this:
 
-```yml
```
Comment (kwinkunks) [inline README.md:36]: The Pygments syntax highlighter doesn't recognize this code block identifier.

File: docs/basic-usage.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Basic usage
```
Comment (kwinkunks) [inline docs/basic-usage.md:1]: This is all directly from the README.

File: docs/commands.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Komodo commands
```
Comment (kwinkunks) [inline docs/commands.md:1]: From the README.

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import sys
+
+# Configuration file for Sphinx documentation builder.
+
+# -- Setup function ----------------------------------------------------------
+
+# Defines custom steps in the process.
+
+
+def autodoc_skip_member(app, what, name, obj, skip, options):
```
Comment (kwinkunks) [inline docs/conf.py:11]: These functions may or may not be needed, they are boilerplate I use for other projects; I find them useful.

**Comment by kwinkunks** [comment]
Added a bit of self-review to help make sense of this large commit.

**Comment by hnformentin** [comment]
Nice move. 
Were you aware of [fmu-docs](https://fmu-docs.equinor.com/) and [fmu-docs-komodo](https://fmu-docs.equinor.com/docs/komodo/equinor_komodo_usage.html)?
The content is different, but what do you think about having both info source at the same place?

File: docs/make.bat [hunk 5]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (xjules) [inline docs/make.bat:1]: What about this `make.bat` file? Isn't it only Windows specific?

Comment (kwinkunks) [reply docs/make.bat:1]: It is; I nuked it in my fork :+1:

**Comment by hnformentin** [comment]
Hei @kwinkunks , I understood that you had updated the PR considering the points that we discussed in the meeting regarding docs.
As I don't see a new commit, would you mind to check the PR and poke when it is read for review again?

**Comment by kwinkunks** [comment]
Thanks @hnformentin -- done! I did not change too much though, and I guess we can't test the GH Workflow until it's merged.

**Review by xjules — APPROVED** [review]
Very nice job @kwinkunks ! I think it can go in now.

**Pull request merged**